docker/recorder/src/main/java/org/arquillian/cube/docker/impl/client/reporter/TakeDockerEnvironment.java

```diff
@@ -0,0 +1,189 @@
+        if (insertedVertex.containsKey(containerId)) {
+            // container is already added, probably because a direct link from another container
+            // now we need to add direct links of this one that before were transitive
+            Object currentContainer = insertedVertex.get(containerId);
+            createDirectLinks(graph, parent, insertedVertex, cubeContainer, currentContainer);
+
+        } else {
+            // create new cube and possible direct link (not transitive ones)
+            Object currentContainer = graph.insertVertex(parent, null, containerId, 0, 0, 80, 30);
+            graph.updateCellSize(currentContainer);
+            insertedVertex.put(containerId, currentContainer);
+
+            createDirectLinks(graph, parent, insertedVertex, cubeContainer, currentContainer);
+        }
+    }
+
+    private void createDirectLinks(mxGraph graph, Object parent, Map<String, Object> insertedVertex, CubeContainer cubeContainer, Object currentContainer) {
+        // create relation to all direct links
+        if (cubeContainer.getLinks() != null) {
+            for (Link link : cubeContainer.getLinks()) {
+                final String linkId = link.getName();
+
+                Object linkContainer = null;
+                if (insertedVertex.containsKey(linkId)) {
+                    linkContainer = insertedVertex.get(linkId);
+                } else {
+                    linkContainer = graph.insertVertex(parent, null, linkId, 0, 0, 80, 30);
+                }
+
+                graph.updateCellSize(currentContainer);
+                graph.insertEdge(parent, null, link.getAlias(), currentContainer, linkContainer);
+                insertedVertex.put(linkId, linkContainer);
+            }
+        }
+    }
+
+    private ScreenshotEntry generateCompositionSchemaImage(mxGraph graph, ReporterConfiguration reporterConfiguration) {
+        final BufferedImage bufferedImage = mxCellRenderer.createBufferedImage(graph, null, 1, Color.WHITE, true, null);
+
+        final File imageFile = new File(createSchemasDirectory(reporterConfiguration.getRootDir()), "docker_composition.png");
+        try {
+            ImageIO.write(bufferedImage, "PNG", imageFile);
+
+            ScreenshotEntry screenshotEntry = new ScreenshotEntry();
+            screenshotEntry.setPhase(When.BEFORE);
+            screenshotEntry.setLink(imageFile.getPath());
+            screenshotEntry.setWidth(bufferedImage.getWidth());
+            screenshotEntry.setHeight(bufferedImage.getHeight());
+            screenshotEntry.setSize(String.valueOf(imageFile.length()));
+            screenshotEntry.setMessage("Docker Composition before executing tests.");
+
+            return screenshotEntry;
+        } catch (IOException e) {
+            log.log(Level.WARNING, String.format("Docker compositions schema could not be generated because of %s.", e));
+        }
+
```

Comment:
BufferedImage has no close method nor implements `AutoClosable` so it is not required